src/main/java/com/amazon/opendistroforelasticsearch/security/support/Base64Helper.java

```diff
@@ -38,79 +44,123 @@
 import java.io.ObjectInputStream;
 import java.io.ObjectOutputStream;
 import java.io.ObjectStreamClass;
+import java.io.OutputStream;
 import java.io.Serializable;
 import java.net.InetAddress;
 import java.net.InetSocketAddress;
 import java.net.SocketAddress;
-import java.util.ArrayList;
+import java.security.PrivilegedAction;
 import java.util.Collection;
+import java.util.Collections;
 import java.util.List;
 import java.util.Map;
+import java.util.Set;
+import java.util.regex.Pattern;
 
 import org.elasticsearch.ElasticsearchException;
+import org.elasticsearch.common.Strings;
 
-import com.amazon.opendistroforelasticsearch.security.resolver.IndexResolverReplacer;
 import com.amazon.opendistroforelasticsearch.security.user.User;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
 import com.google.common.io.BaseEncoding;
 
 public class Base64Helper {
 
-    public static String serializeObject(final Serializable object) {
+    private static final Set<Class<?>> SAFE_CLASSES = ImmutableSet.of(
+        String.class,
+        SocketAddress.class,
+        InetSocketAddress.class,
+        Pattern.class,
+        User.class,
+        SourceFieldsContext.class,
+        LdapUser.class,
+        SearchEntry.class,
+        LdapEntry.class,
+        AbstractLdapBean.class,
+        LdapAttribute.class
+    );
+
+    private static final List<Class<?>> SAFE_ASSIGNABLE_FROM_CLASSES = ImmutableList.of(
```

Comment:
Should this be a Set as well?

Reply:
It can be `List` or `Set` as there is no lookup by key support required and elements needs to be unique. As it is statically populated and we can guarantee uniqueness, `List` or even `Collection` is sufficient.